### PR TITLE
added task for analysis challege: used for TPC trk efficiency of prim…

### PR DESCRIPTION
**Pull request by iravasen**
…ary el,pi,ka,p and DCAxy distributions of primaries, secondaries

### CODE REVIEW

**Comment by iravasen** [comment]
Hello @jgrosseo,
this is my first PR in O2, probably that's why one of the test fails. Should I do something on my side? 

Thanks

File: Analysis/Tasks/trackchecks.cxx [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    } else {
+      LOGF(info, "isMC=%d - Running on DATA", isMC.value);
+    }
+
+    //event selection
+    if (!isMC && !col.alias()[kINT7]) // trigger (should be skipped in MC)
+      return;
+    if (!col.sel7()) //additional cuts
+      return;
+    if (abs(col.posZ()) > 10.) // |v_z|<10cm
```
Comment (jgrosseo) [inline Analysis/Tasks/trackchecks.cxx]: You can use a Filter for this

File: Analysis/Tasks/trackchecks.cxx [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        if (abs(y) < 0.5) {
+          hDCAxyReco_truepid->Fill(ptype, track.pt(), dcas.first);
+          if (isPrimary)
+            hDCAxyPrim_truepid->Fill(ptype, track.pt(), dcas.first);
+          else
+            hDCAxySeco_truepid->Fill(ptype, track.pt(), dcas.first);
+        }
+      }
+
+      //Additional CUTS
```
Comment (jgrosseo) [inline Analysis/Tasks/trackchecks.cxx]: Why you don't use the TrackSelection class, or even better the track selection workflow and then you can use a Filter to only get accepted tracks in your task

File: Analysis/Tasks/trackchecks.cxx (hunk 1 above)
Comment (iravasen) [reply Analysis/Tasks/trackchecks.cxx]: I put the vertex_z selection into a filter. I do not know how to put the others cuts, I tried different possibilities but it doesn't compile.

File: Analysis/Tasks/trackchecks.cxx (hunk 2 above)
Comment (iravasen) [reply Analysis/Tasks/trackchecks.cxx]: I wanted to use the TrackSelection class but I see that aod::TracksExtended (to get dcaXY and dcaZ from it directly) is not in the Pb-Pb MC I'm using for testing (converted on Aug. 5th). Then, if possible I would like to keep the track selection as it is in order to play a bit with the cuts in a easier way. My objective is to compare TPC tracking efficiency in the new and old (Run2) framework.

Comment (iravasen) [reply Analysis/Tasks/trackchecks.cxx]: It's also very nice to use in the near future the trackselection workflow but I didn't find instructions for it.

Comment (jgrosseo) [reply Analysis/Tasks/trackchecks.cxx]: You can join the hyperloop channel for the analysis challenge and ask there. The track selection workflow will add the table you were missing. In a nutshell you just add o2-analysis-trackselection to your workflow.

Comment (iravasen) [reply Analysis/Tasks/trackchecks.cxx]: ok, thanks! I'm already in the channel, I will ask for more details there. In the meantime I would like to use this track selection if possible as anticipated in the previous message.

Comment (jgrosseo) [reply Analysis/Tasks/trackchecks.cxx]: I don't agree with this approach. The code is more or less copy-pasted from the TrackSelection class. It is better to use the TrackSelection class (no need to use the default cuts), and configure them as you like. You should be able to do any cut you like. @mkruegerGitHub may comment as well.

**Review by jgrosseo — CHANGES_REQUESTED** [review]
See comments

File: Analysis/Tasks/trackchecks.cxx (hunk 2 above)
Comment (iravasen) [reply Analysis/Tasks/trackchecks.cxx]: ok, this will require some time since I would like to test few things. I'll be back to you asap.

Comment (mario-krueger) [reply Analysis/Tasks/trackchecks.cxx]: I also believe that we should avoid duplicating code that already exists esewhere. Documenation for the track selection is already in preparation.
In addition, the code for calculating the dca that was copied here is not entirely correct (see https://github.com/AliceO2Group/AliceO2/pull/4450).

Comment (iravasen) [reply Analysis/Tasks/trackchecks.cxx]: thanks, the documentation is really appreciated.

File: Analysis/Tasks/trackchecks.cxx [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    hDCAxySeco_truepid->Add(4, TH2F("hDCAxySeco_truepid_de", "DCAxy secondaries (true De); #it{p}_{T} (GeV/#it{c}); DCAxy (cm)", NPTBINS, pt_bins, 800, -4., 4.));
+
+    fCutDCAxy = new TF1("fMaxDCAxy", "[0]+[1]/(x^[2])", 0, 1e10); //from TPC analysis task (Run2)
+    fCutDCAxy->SetParameter(0, 0.0105);
+    fCutDCAxy->SetParameter(1, 0.0350);
+    fCutDCAxy->SetParameter(2, 1.1);
+  } //init
+
+  //Filters
+  Filter collfilter = nabs(aod::collision::posZ) < (float)10.;
```
Comment (aalkin) [inline Analysis/Tasks/trackchecks.cxx]: You can use `10.f` notation here, instead of a type cast.

File: Analysis/Tasks/trackchecks.cxx (hunk 2 above)
Comment (mario-krueger) [reply Analysis/Tasks/trackchecks.cxx]: Hi Ivan,
the documentation is now available. You can find it here: https://aliceo2group.github.io/analysis-framework/tracksel.html.
Cheers,
  Mario

File: Analysis/Tasks/trackchecks.cxx (hunk 3 above)
Comment (iravasen) [reply Analysis/Tasks/trackchecks.cxx]: done, at the end I made configurable cut.

File: Analysis/Tasks/trackchecks.cxx (hunk 2 above)
Comment (iravasen) [reply Analysis/Tasks/trackchecks.cxx]: THanks a lot Mario, I've implemented it.

**Comment by iravasen** [comment]
Hello all, I'm done with the modifications. When you have time you can check the code. Thanks a lot.

**Comment by jgrosseo** [comment]
This is fine with me. The only item left is that it should go to some sub folder of Tracks? Does this belong to PWGPP or to which activity?

**Comment by iravasen** [comment]
Thanks for the quick feedback. Yes, better to move it to the PWGLF folder. Doing it.

**Review by jgrosseo — APPROVED** [review]

**Pull request merged**